### ISSUE TITLE
matchesValue allows any value for BigInt

### ISSUE THREAD

**Issue by bogeeee** [typescript-rtti/typescript-rtti#99]
```
import { reflect } from 'typescript-rtti';

class A {
    i: BigInt
}

console.log(reflect(A).getProperty("i").matchesValue("anyvalue"));
```

returns true. Expected: false

**Comment by bogeeee** [comment]
Can see this issue only in the (nice) [playground](https://typescript-rtti.org/#fFYyfHsic291cmNlIjoiaW1wb3J0IHsgcmVmbGVjdCB9IGZyb20gJ3R5cGVzY3JpcHQtcnR0aSc7XG5cbmNsYXNzIEEge1xuICAgIGk6IEJpZ0ludFxufVxuY29uc3QgcHJvcCA9IHJlZmxlY3QoQSkuZ2V0UHJvcGVydHkoXCJpXCIpO1xuY29uc29sZS5sb2cocHJvcC5tYXRjaGVzVmFsdWUoXCJhbnl2YWx1ZVwiKSk7IiwiY29tcGlsZXJPcHRpb25zIjoiIn0=)

but tested on my local machine with mentioned versions: "Uses typescript-rtti@0.8.3 and typescript@4.7.4" and it works properly. So maybe it's a non bug.

**Comment by rezonant** [comment]
Yep, this is working in real environments, but in Playground the type of `i` is being emitted as `any` which is causing this issue.

```
[1849]: { RΦ: t => ({ TΦ: "~" }) }
```

(`~` is `T_ANY` in the RTTI format)
